packages/experimental/Button/src/Button.tsx

```diff
@@ -1,36 +1,44 @@
 /** @jsx withSlots */
 import * as React from 'react';
-import { Image, View } from 'react-native';
+import { View } from 'react-native';
 import { buttonName, ButtonType, ButtonProps } from './Button.types';
 import { Text } from '@fluentui-react-native/experimental-text';
 import { stylingSettings } from './Button.styling';
-import { filterImageProps } from '@fluentui-react-native/adapters';
 import { compose, mergeProps, withSlots, UseSlots } from '@fluentui-react-native/framework';
 import { useButton } from './useButton';
+import { Icon } from '@fluentui-react-native/icon';
+import { createIconProps } from '@fluentui-react-native/interactive-hooks';
 
 export const Button = compose<ButtonType>({
   displayName: buttonName,
   ...stylingSettings,
   slots: {
     root: View,
-    icon: Image,
+    icon: Icon,
     content: Text,
   },
-  filters: {
-    icon: filterImageProps,
-  },
   render: (userProps: ButtonProps, useSlots: UseSlots<ButtonType>) => {
     const button = useButton(userProps);
+    const iconProps = createIconProps(userProps.icon);
     // grab the styled slots
-    const Slots = useSlots(userProps, (layer) => button.state[layer] || userProps[layer]);
+    const Slots = useSlots(userProps, layer => button.state[layer] || userProps[layer]);
     // now return the handler for finishing render
     return (final: ButtonProps, ...children: React.ReactNode[]) => {
       const { icon, content, ...mergedProps } = mergeProps(button.props, final);
-
+      const marginBetween = {
+        marginLeft: 0,
+      };
+      if (icon && content) {
+        marginBetween.marginLeft = 10;
+      }
```

Comment:
Is there a margin you can put into root that provides spacing between it's children instead of doing this conditional margin? Something like alignContent https://reactnative.dev/docs/flexbox#align-content ?

Reply:
No, because the root is only wide enough for the content and padding that is hard coded in. Using justifyContent (since it is in 'row' direction), there is no space to do space-around which seems like the closest thing I would want. Essentially everything is packed in as tight as can be without extra spacing to move things around. If there was a fixed width of 100 let's say and padding, icon, and content only takes up 60 out of that 100, the rest (40) would be left over for spacing adjustments.

Reply:
OK, if we go this route, I would go with this suggestion to make it more of a const and less a mutable variable.
I'll think a bit more today if there's a better way to modify the style today. Otherwise the rest of the change looks good!

```suggestion
      const marginBetween = {
        marginLeft: (icon && content) ? 10 : 0,
      };
```